### PR TITLE
[OSSA] Migrate AccessedStorage

### PR DESCRIPTION
**Pull request by atrick**
A couple of minor changes to the AccessedStorage utilities.

Handle begin_borrow->convert_escape_to_noescape

All tests that directly use the utilities are migrated.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci test

**Review by meg-gupta — APPROVED** [review]
LGTM. I just had a comment on whether copy_value needs to be handled as well while looking for callee origin.

File: include/swift/SIL/SILInstruction.h [hunk 1]
```diff
@@ -9048,6 +9048,11 @@ SILValue ApplyInstBase<Impl, Base, false>::getCalleeOrigin() const {
       Callee = CETN->getOperand();
       continue;
     }
+    // convert_escape_to_noescape's are within a borrow scope.
+    if (auto *beginBorrow = dyn_cast<BeginBorrowInst>(Callee)) {
+      Callee = beginBorrow->getOperand();
+      continue;
+    }
```
[the remark this answers is not in the document]
Comment (atrick) [reply include/swift/SIL/SILInstruction.h:9055]: yes, we should handle `copy_value`. I added another commit to this PR.

**Comment by atrick** [comment]
@swift-ci test

**Pull request merged**